packages/core/chrome/core-chrome-browser-internal/src/project_navigation/project_navigation_service.ts

```diff
@@ -169,6 +188,68 @@ export class ProjectNavigationService {
     };
   }
 
+  private initNavigation(
+    navTreeDefinition: Observable<NavigationTreeDefinition>,
+    { navLinksService, cloudUrls }: { navLinksService: ChromeNavLinks; cloudUrls: CloudURLs }
+  ) {
+    if (this.navigationTree$.getValue() !== undefined) {
+      throw new Error('Project navigation has already been initiated.');
+    }
+
+    const deepLinksMap$ = navLinksService.getNavLinks$().pipe(
+      map((navLinks) => {
+        return navLinks.reduce((acc, navLink) => {
+          acc[navLink.id] = navLink;
+          return acc;
+        }, {} as Record<string, ChromeNavLink>);
+      })
+    );
+
+    const cloudLinks = getCloudLinks(cloudUrls);
+
+    this.navigationTree$
+      .pipe(
+        takeUntil(this.stop$),
+        filter((navTree): navTree is ChromeProjectNavigationNode[] => Boolean(navTree))
+      )
+      .subscribe((navTree) => {
+        this.projectNavigationNavTreeFlattened = flattenNav(navTree);
```

Comment:
can we move these two side effects inside the `subscribe` below? So that setting navigation tree state is in one place

Reply:
Great suggestion 👍 Cleaned it up in https://github.com/elastic/kibana/pull/174544/commits/bcb2bb3c87e5a2c6b5e895ea422601c743e6d935